timemachine/fe/single_topology.py

```diff
@@ -1135,6 +1148,34 @@ def assert_bonds_defined_for_chiral_volumes(vacuum_system, bond_k_min=DEFAULT_BO
                 raise MissingBondsInChiralVolumeException(f"bond {(c, k)} missing from Chiral Volume {(c, i, j, k)}")
 
 
+def assert_torsions_defined_over_non_linear_angles(vacuum_system):
+    """
+    Assert that torsions are never defined over angle terms with an equilibrium value close to 180.
+    """
+    linear_angles = set()
+
+    for idxs, angle_params in zip(vacuum_system.angle.potential.idxs, vacuum_system.angle.params):
+        angle_k, angle_a0 = angle_params[0], angle_params[1]
+
+        if angle_k > 0:
+            if abs(angle_a0 - np.pi) < 0.174533:  # 10 degrees, arbitrary but conservative threshold
+                linear_angles.add(tuple(idxs))
+
+    for (i, j, k, l), (proper_k, _, _) in zip(vacuum_system.proper.potential.idxs, vacuum_system.proper.params):
+        if proper_k > 0:
+            if canonicalize_bond((i, j, k)) in linear_angles:
+                raise TorsionsDefinedOverLinearAngleException(f"angle {(i,j,k)} is linear in torsion {(i,j,k,l)}")
```

Comment:
nit: Would be nice to specify whether the torsion is proper/improper in the warnings to make debugging in the future.

Reply:
good call - will fix.